### PR TITLE
Infinite "Processing" State for Invalid URLs

### PR DESCRIPTION
**Pull request by Aryanfour5**
Handling #12 

 This pull request introduces several enhancements and features to the Paste Viewer component, aimed at improving user experience and functionality.

Changes Made:

Error Handling:

Added comprehensive error handling when fetching paste data from the API. If the paste ID is invalid or non-existent, a user-friendly error message is displayed.
The error message prompts users to check the URL and informs them that the paste may have expired or been removed.

### CODE REVIEW

File: src/routes/[id]/+page.svelte [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
-      .then((data) => {
-        // console.log("Fetched paste:", data.id);
-        paste = data.id;
-        new ClipboardJS(".btn-clip");
-      })
-      .catch((error) => {
-        console.log("Error fetching paste:", error);
-      });
+    id = $page.params.id;
+    console.log(`fetching /api/paste/${id}`);
```
Comment (onkar69483) [inline src/routes/[id]/+page.svelte:18]: let the route be same as old route i.e /api?id=${id}

File: src/routes/[id]/+page.svelte [hunk 2]
```diff
@@ -1,29 +1,36 @@
...
+
+    try {
+      const res = await fetch(`/api/paste/${id}`);
+      if (!res.ok) {
+        throw new Error("Invalid or non-existent paste ID.");
+      }
+      const data = await res.json();
+      paste = data;
+      code = paste.text;
```
Comment (onkar69483) [inline src/routes/[id]/+page.svelte:27]: The old method was okay here 
fetch(`/api?id=${id}`)
      .then((res) => res.json())
      .then((data) => {
        // console.log("Fetched paste:", data.id);
        paste = data.id;
        new ClipboardJS(".btn-clip");
      })

File: src/routes/[id]/+page.svelte (hunk 1 above)
Comment (Aryanfour5) [reply src/routes/[id]/+page.svelte:18]: ok

**Review by onkar69483 — COMMENTED** [review]
1. Let the route be /{id} as it is used in all other files too, if you change it here as /paste/{id} you would have to change it in every file wherever it is used, so its better to be kept as /{id}

2. Now normal pastes are not opening they are showing paste not found error which should not happen. We only need to show error for the paste which is not found
image

**Comment by onkar69483** [comment]
For invalid URL it is showing properly

image


But for valid url the paste is not loading / it is showing blank

image

Instead it should show something like this

image

File: src/routes/[id]/+page.svelte [hunk 3]
```diff
@@ -1,31 +1,41 @@
...
+
+    try {
+      // Fetching the paste using the updated route
+      const res = await fetch(`/api?id=${id}`);
+      if (!res.ok) {
+        throw new Error("Invalid or non-existent paste ID.");
+      }
+      const data = await res.json();
+      paste = data;
+      code = paste.text;
```
Comment (onkar69483) [inline src/routes/[id]/+page.svelte:29]: Let it be paste.id and not paste.text

File: src/routes/[id]/+page.svelte [hunk 4]
```diff
@@ -68,81 +73,79 @@
...
           }}
           class="btn-clip bg-green-600 hover:bg-lime-600 text-white font-medium py-2 px-4 mt-3 rounded-md focus:outline-none focus:ring-2 focus:ring-indigo-500"
-          data-clipboard-text={code}
+          data-clipboard-text={paste.text}
         >
           COPY TEXT
         </button>
         <div>
-          <!-- download pdf button -->
           <button
```
Comment (onkar69483) [inline src/routes/[id]/+page.svelte:135]: Use old button theme for download and share 

image

**Pull request merged**